invokeai/app/services/session_queue/session_queue_sqlite.py

```diff
@@ -145,7 +144,9 @@ async def enqueue_batch(self, queue_id: str, batch: Batch, prepend: bool) -> Enq
                     (batch.batch_id,),
                 )
                 item_ids = [row[0] for row in cursor.fetchall()]
+            self._conn.commit()
         except Exception:
+            self._conn.rollback()
```

Comment:
The pattern established elsewhere in this code is that we roll back, so this line appears to still be relevant.

Reply:
We could certainly trim a bunch of lines of code since transactions are automatically rolled back on exception.

Reply:
Sometimes you need to do other exception handling, can't use the context manager everywhere. But generally, yes, we could use it more. I don't know if there's a particular reason not to in most places.